File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/token/LoginLog.java
package pers.acp.admin.oauth.token;

/**
 * @author zhang by 08/03/2019
 * @since JDK 11
 */
public class LoginLog {

    public String getAppid() {
        return appid;
    }

    public void setAppid(String appid) {
        this.appid = appid;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getUserid() {
        return userid;
    }

    public void setUserid(String userid) {
        this.userid = userid;
    }

    private String appid;

    private String date;

    private String userid;

}

File: cloud/log-server/src/main/kotlin/pers/acp/admin/log/entity/LoginLog.kt
package pers.acp.admin.log.entity

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import org.hibernate.annotations.GenericGenerator
import pers.acp.admin.log.base.BaseLogEntity
import io.github.zhangbinhub.acp.core.CommonTools
import io.github.zhangbinhub.acp.core.task.timer.Calculation

import javax.persistence.*

/**
 * @author zhang by 21/05/2019
 * @since JDK 11
 */
@Entity
@Table(name = "t_log_login",
        uniqueConstraints = [UniqueConstraint(columnNames = ["logId", "requestTime"])],
        indexes = [
            Index(columnList = "requestTime"),
            Index(columnList = "userId"),
            Index(columnList = "clientId,loginDate"),
            Index(columnList = "userId,clientId,loginDate")
        ])
@org.hibernate.annotations.Table(appliesTo = "t_log_login", comment = "登录日志")
@ApiModel("登录日志")
data class LoginLog(
        @Id
        @GenericGenerator(name = "idGenerator", strategy = "uuid")
        @GeneratedValue(generator = "idGenerator")
        @Column(length = 36, nullable = false)
        @ApiModelProperty("ID")
        var id: String = "",

        @Column(nullable = false)
        @ApiModelProperty("登录日期")
        var loginDate: String = CommonTools.getDateTimeString(null, Calculation.DATE_FORMAT)
) : BaseLogEntity()